app/scripts/controllers/selfservice/CreateSelfServiceUserController.js

```diff
@@ -0,0 +1,98 @@
+(function (module) {
+    mifosX.controllers = _.extend(module, {
+        CreateSelfServiceUserController: function (scope, resourceFactory, routeParams,$uibModal) {
+            // TODO: Add Account Is Active Logic through email validation
+            scope.accountActive = false;    
+            scope.available = [];
+            scope.selected = [];
+            scope.selectedRoles = [] ;
+            scope.availableRoles = [];
+            scope.formData = {
+                isSelfServiceUser: true,
+                sendPasswordToEmail: true,
+                roles: []
+            };
+            scope.clientId = routeParams.clientId;
+            resourceFactory.userTemplateResource.get(function (data) {
+                scope.availableRoles = data.availableRoles;
```

Comment:
I believe we need to have a separate API to retrieve only self service user roles so that we can filter out other roles.